### PR TITLE
Add browser OAuth sign-in for OpenRouter in CLAI

### PR DESCRIPTION
**Pull request by mpfaffenberger**
## Summary
- Add **Sign in with browser** alongside masked API-key entry under `/model > openrouter`.
- Use OpenRouter authorization-code + S256 PKCE, an ephemeral loopback callback, and pasted callback URLs/codes for remote sessions.
- Reuse existing discovery and credential storage. Save only after model selection; keep existing credentials on failure or cancellation.
- Bound login and HTTP waits, clean up listener/prompt tasks, and avoid exposing exchange response bodies.
- Document the flow in `README.md` and `PLUGINS.md`.

## Sources
- Protocol: https://openrouter.ai/docs/use-cases/oauth-pkce
- Local reference inspected: `~/code/code_puppy_core_plugins/code_puppy_core_plugins/openrouter_oauth/oauth_flow.py`. This implementation uses CLAI async IO and credential storage rather than copying the threaded Code Puppy implementation.

## Verification
- Repository-wide Ruff lint and format checks passed.
- Strict Pyright on changed source/tests passed.
- 68 focused tests passed across OpenRouter, browser auth, model menu, and Codex auth.
- Tests use a real loopback callback server and mocked OpenRouter HTTP responses. No live account authorization was performed.

- CI confirmed 100% CLAI branch coverage after adding callback edge-case tests.

### CODE REVIEW

File: pydantic-clai2/PLUGINS.md [hunk 1]
```diff
@@ -386,6 +386,8 @@ Open `/model`, choose `vllm`, then enter a trusted HTTP(S) server root or `/v1`
 
 ## openrouter connection
 
-Open `/model`, choose `openrouter`, then paste an API key from https://openrouter.ai/keys in the masked prompt, then select a model from the live catalog. CLAI validates the key with `/api/v1/key` before fetching `/api/v1/models`. This flow uses API-key authentication, not browser OAuth.
+Open `/model`, choose `openrouter`, then choose **Sign in with browser** or **Enter API key**. Browser sign-in opens OpenRouter's [PKCE authorization flow](https://openrouter.ai/docs/use-cases/oauth-pkce) and receives a user-controlled API key on a temporary loopback listener. If the browser cannot reach CLAI (for example over SSH), paste the final callback URL or authorization code into the terminal. If no browser opens, open the printed authorization URL manually. Login times out after five minutes; Ctrl-C cancels it. You can revoke the generated key on OpenRouter.
```
[the remark this answers is not in the document]
Comment (mpfaffenberger) [reply pydantic-clai2/PLUGINS.md]: Fixed in both README.md and PLUGINS.md. The docs now distinguish the loopback authorization code from the API key returned by the HTTPS exchange.

**Comment by mpfaffenberger** [comment]
Acknowledged the Approvability assessment: this authentication change should receive human review before merge. No merge is requested here. The loopback callback carries only an authorization code; key exchange is HTTPS with S256 PKCE, and persistence uses the existing credential store only after model selection. CLAI tests on Python 3.11/3.14 and its 100% coverage gate are green.
